transports/janus_websockets.c

```diff
@@ -776,7 +774,8 @@ static void janus_websockets_destroy_client(
 	ws_client->buflen = 0;
 	ws_client->bufpending = 0;
 	ws_client->bufoffset = 0;
-	janus_mutex_unlock(&ws_client->mutex);
+	janus_mutex_unlock(&ws_client->ts->mutex);
+	janus_transport_session_destroy(ws_client->ts);
```

Comment:
`janus_request_new` increases the refcount for `instance` (`janus_transport_session`) and `janus_websockets_send_message` checks whether `transport_p` is `NULL` inside the mutex, so that my original solution should be fine.